x-pack/test/detection_engine_api_integration/security_and_spaces/group1/update_actions.ts

```diff
@@ -120,10 +120,27 @@ export default ({ getService }: FtrProviderContext) => {
           rule_id: immutableRule.rule_id, // Rule id should match the same as the immutable rule
           version: immutableRule.version, // This version number should not change when an immutable rule is updated
           immutable: true, // It should stay immutable true when returning
+          required_fields: immutableRule.required_fields, // required_fields cannot be modified, so newRuleToUpdate will have required_fields from immutable rule
```

Comment:
Why `required_fields` are special so that we need to check them explicitly? What about other rule fields like `related_integrations`, `index`, etc? It feels like this `expected` object should be built based on `immutableRule`. Something like this:

```ts
        const expected = {
          ...immutableRule,
          immutable: true,
          actions: ...
          // and other properties that we expect to be changed
```

Reply:
>  It feels like this expected object should be built based on immutableRule
Agree on that. Here is a quick explanation why it's done like this

Scenarios written in this file are using immutable rule, fetching it by hardcoded ruleId.
```
       const immutableRule = await getRule(supertest, log, '9a1a2dae-0b5f-4c3d-8305-a268d404c306');
        const hookAction = await createNewAction(supertest, log);
```
Then rule mock data is created from `getSimpleRule`, that uses immutable rule id,
Existing immutable rule is getting updated with that mock, and then assertions are made against it.

```
        const newRuleToUpdate = getSimpleRule(immutableRule.rule_id);
        const ruleToUpdate = getRuleWithWebHookAction(hookAction.id, false, newRuleToUpdate);
        const updatedRule = await updateRule(supertest, log, ruleToUpdate);
        const bodyToCompare = removeServerGeneratedProperties(updatedRule);

```

I'm not sure, what is rationale for doing this. But I didn't want for a fix that unblocks another PR, start refactoring tests there, given BC4 soon.

We may look further, after release, as current implementation is not obvious for me, at least.
If you prefer I can look to refactor it now, but I'm not sure if no side effects will be discovered

Reply:
@vitaliidm We should definitely unblock https://github.com/elastic/kibana/pull/138574 as soon as we can, so yea, let's merge this PR as is.

However, I think we should fix this test and make it more robust so it doesn't fail in the future for similar reasons. The fix could be safely backported to the `8.4` branch because it's a test 🙂 We don't have to wait for the release to happen, the test can be reworked in a follow-up PR right away.

Reply:
thanks @banderror 
Here is a ticket to track it: https://github.com/elastic/kibana/issues/138757
I think it would make to schedule it the next release, so next prebuilt update won't cause similar failures